### DISCUSSION TITLE
MorphTo fields won't work when morph type column uses php Enum with casts

### DISCUSSION THREAD

**Discussion by andypooletrioteca** [laravel/nova-issues#5306] [Q&A] [answered]
- Laravel Version: 9.48.0
- Nova Version: v4.21.0 (Silver Surfer)
- PHP Version: 8.2.1

### Description:
MorphTo field doesn't seem to work well when a model has the type column casted as a php Enum.

### Detailed steps to reproduce the issue on a fresh Nova installation:

Have an enum defining the models that can be contactable.
```
enum Contactable: string
{
    case PROVIDER => 'App\Models\Provider',
    case CUSTOMER => 'App\Models\Customer',
}
```

Have a polymorphic relation to these contactable models
```
class Contact extends Model
{
    protected $casts = [
        'contactable_type' => Contactable::class,
    ];
    
    public function contactable(): MorphTo
    {
        return $this->morphTo();
    }
}
```

Define a nova resource for this Model

```
class Contact extends Resource
{
    public static $model = 'App\Models\Contact';
    public function fields(NovaRequest $request)
    {
        return [
            MorphTo::make('Contact', 'contactable')->types(['App/Nova/Customer', 'App/Nova/Provider']),
        ];
    }
}
```

Then you get:
```
{
    "message": "Illegal offset type",
    "exception": "TypeError",
    "file": "/var/www/html/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Relations/Relation.php",
    "line": 475,
    "trace": [
        {
            "file": "/var/www/html/vendor/laravel/nova/src/Fields/MorphTo.php",
            "line": 276,
            "function": "getMorphedModel",
            "class": "Illuminate\\Database\\Eloquent\\Relations\\Relation",
            "type": "::"
        },
        ...
```

I guess contactable_type is not expected to be casted as an Enum... but why not?

**Comment by crynobone** [comment, marked as answer]
> I guess contactable_type is not expected to be casted as an Enum... but why not?

If you want to do this, that would mean that every usage of `contactable_type` need to pass the Enum instance while Nova POST requests can only send in a `string`. This is currently not supported add just complicates the usage without any advantage.

You can set/limit the possible type by using `MorphTo::types()` as suggested in the documentation: https://nova.laravel.com/docs/4.0/resources/relationships.html#morphto
